contracts/contracts/strategies/NativeStaking/CompoundingValidatorManager.sol

```diff
@@ -840,6 +882,114 @@ abstract contract CompoundingValidatorManager is Governable {
             }
         }
 
+        uint256 depositsCount = depositsRoots.length;
+        uint256 totalDepositsWei = 0;
+
+        // If there are no deposits then we can skip the deposit verification
+        // This section is after the validator balance verifications so an exited validator will be marked
+        // as EXITED before the deposits are verified. If there was a deposit to an exited validator
+        // then the deposit can only be removed once the validator is fully exited.
+        if (depositsCount > 0) {
+            // Verify the slot of the first pending deposit matches the beacon chain
+            bool isDepositQueueEmpty = IBeaconProofs(BEACON_PROOFS)
+                .verifyFirstPendingDeposit(
+                    blockRoot,
+                    firstPendingDeposit.slot,
+                    firstPendingDeposit.pubKeyHash,
+                    firstPendingDeposit.pubKeyProof
+                );
+
+            // If there are no deposits in the beacon chain queue then our deposits must have been processed.
+            // If the deposits have been processed, each deposit will need to be verified with `verifyDeposit`
+            require(!isDepositQueueEmpty, "Deposits have been processed");
+
+            // The verification of the validator the first pending deposit is for must be on or after when
+            // `snapBalances` was called.
+            require(
+                balancesMem.timestamp <= validatorVerificationBlockTimestamp,
```

Comment:
A thought: what if we remove the `validatorVerificationBlockTimestamp` and just calculate it as: `balancesMem.timestamp + SLOT_DURATION`. This would impose an extra limitation to `verifyBalances` where 3 slots in succession are required to produce a block instead of 2.